boards: stm32: add nucleo board stm32g474re

FRASTM opened the pull request:
This patch adds the nucleo_g474re board
     based on the stm32g474re from STMicroelectronics.

Signed-off-by: Francois Ramu <francois.ramu@st.com>

FRASTM commented:
add Kconfig.defconfig.stm32g474re with soc

erwango commented on dts/arm/st/g4/stm32g474.dtsi:
```diff
@@ -0,0 +1,7 @@
+/*
+ * Copyright (c) 2019 Richard Osterloh <richard.osterloh@gmail.com>
```
Pleae use your copyright.

dbkinder replied:
Can't just delete Richard's copyright, but you can add Francois'.

erwango commented on dts/arm/st/g4/stm32g474Xe.dtsi:
```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright (c) 2019 Richard Osterloh <richard.osterloh@gmail.com>
```
Ditto

erwango commented on soc/arm/st_stm32/stm32g4/Kconfig.defconfig.stm32g474re:
```diff
@@ -0,0 +1,21 @@
+# Kconfig - STMicroelectronics STM32G474RE MCU
+#
+# Copyright (c) 2019 Richard Osterloh <richard.osterloh@gmail.com>
```
ditto

erwango commented on dts/arm/st/g4/stm32g4.dtsi:
```diff
@@ -415,6 +415,18 @@
 			};
 		};
 
+		lptim1: timers@40007c00 {
```
Please remove for now.

erwango commented on dts/arm/st/g4/stm32g474.dtsi:
```diff
@@ -0,0 +1,7 @@
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <st/g4/stm32g4.dtsi>
```
#include <st/g4/stm32g431.dtsi>

erwango commented on soc/arm/st_stm32/stm32g4/Kconfig.defconfig.stm32g474re:
```diff
@@ -0,0 +1,21 @@
+	int
+	default 102
+
+if GPIO_STM32
```
Remove these lines

erwango commented on soc/arm/st_stm32/stm32g4/dts_fixup.h:
```diff
@@ -260,4 +260,11 @@
 #define DT_WWDT_0_CLOCK_BITS			DT_INST_0_ST_STM32_WINDOW_WATCHDOG_CLOCK_BITS
 #define DT_WWDT_0_CLOCK_BUS			DT_INST_0_ST_STM32_WINDOW_WATCHDOG_CLOCK_BUS
 
+#define DT_LPTIM_1_BASE_ADDRESS	DT_ST_STM32_TIMERS_40007C00_BASE_ADDRESS
```
Remove for now.

erwango commented on soc/arm/st_stm32/stm32g4/soc.h:
```diff
@@ -78,6 +78,11 @@
 #include <stm32g4xx_ll_pwr.h>
 #endif
 
+#ifdef CONFIG_STM32_LPTIM_TIMER
```
Please remove for now

erwango commented on boards/arm/nucleo_g431rb/nucleo_g431rb.dts:
```diff
@@ -89,6 +89,10 @@
 	status = "okay";
 };
 
+&lptim1 {
```
Please remove for now.

erwango commented on boards/arm/nucleo_g431rb/nucleo_g431rb_defconfig:
```diff
@@ -3,7 +3,6 @@
 CONFIG_ARM=y
 CONFIG_SOC_SERIES_STM32G4X=y
 CONFIG_SOC_STM32G431XX=y
-CONFIG_CORTEX_M_SYSTICK=y
```
Change not related to current PR (different board)

FRASTM committed:
soc: arm: st_stm32: configure stm32g431 series

This patch cleans the defconfig of the STM32G431
     device from STMicroelectronics.

Signed-off-by: Francois Ramu <francois.ramu@st.com>

FRASTM commented:
change fater review
set stm32g474 flash size to 512K and 128K of SRAM

erwango commented on boards/arm/nucleo_g474re/nucleo_g474re_defconfig:
```diff
@@ -0,0 +1,59 @@
+CONFIG_ARM=y
+CONFIG_SOC_SERIES_STM32G4X=y
+CONFIG_SOC_STM32G474XX=y
+CONFIG_CORTEX_M_SYSTICK=y
```
Remove this line

FRASTM replied:
CONFIG_CORTEX_M_SYSTICK=y removed

erwango commented on dts/arm/st/g4/stm32g474.dtsi:
```diff
@@ -0,0 +1,8 @@
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <st/g4/stm32g4.dtsi>
```
Remove this line

FRASTM replied:
line "#include <st/g4/stm32g4.dtsi>" replaced by "#include <st/g4/stm32g431.dtsi>"

erwango replied:
I still see both lines actually

dbkinder commented on boards/arm/nucleo_g474re/doc/index.rst:
```diff
@@ -0,0 +1,241 @@
+Default Zephyr Peripheral Mapping:
+----------------------------------
+
+- UART_1_TX : PC4
```
This long list will display as a multi-column list if you add this directive and a blank line before the first list item:
```
.. rst-class:: rst-columns

- UART_1_TX : PC4
- UART_1_RX : PC5
  ...
```

FRASTM replied:
ok, could also fit  nucleo_g431rb/

dbkinder reviewed (changes requested):
doc changes look fine, one suggested edit to make the generated HTML look better.

FRASTM commented:
change rst file in doc for both boards nucleo_g474re and nucleo_g431rb

FRASTM commented:
change rst doc for both nucleo_g744re and nucleo_g431rb boards

erwango commented on boards/arm/nucleo_g474re/Kconfig.board:
```diff
@@ -0,0 +1,11 @@
+# Kconfig - STM32G474RE Nucleo board configuration
+#
+# Copyright (c) 2019 Richard Osterloh <richard.osterloh@gmail.com>
```
Remove this line

erwango commented on boards/arm/nucleo_g474re/Kconfig.defconfig:
```diff
@@ -0,0 +1,45 @@
+# Kconfig - STM32G474RE Nucleo board configuration
+#
+# Copyright (c) 2019 Richard Osterloh <richard.osterloh@gmail.com>
```
Remove this line

erwango commented on boards/arm/nucleo_g474re/nucleo_g474re_defconfig:
```diff
@@ -3,7 +3,6 @@
 CONFIG_ARM=y
 CONFIG_SOC_SERIES_STM32G4X=y
 CONFIG_SOC_STM32G474XX=y
-CONFIG_CORTEX_M_SYSTICK=y
```
Squash this into board commit

erwango commented on boards/arm/nucleo_g474re/doc/index.rst:
```diff
@@ -133,6 +133,8 @@ For mode details please refer to `STM32G4 Nucleo-64 board User Manual`_.
 Default Zephyr Peripheral Mapping:
 ----------------------------------
 
+.. rst-class:: rst-columns
```
Squash this into board commit

erwango reviewed (changes requested):
Please rebase onto latest master as current version can't build (due to missing Cortex Systick activation).

FRASTM commented:
rebasing on HEAD

FRASTM commented:
Squash doc rst file  into board commit

FRASTM commented:
line with wrong Copyright removed

FRASTM committed:
soc: arm: st_stm32: add stm32g474 series

This patch introduces the support of the STM32G474
     device from STMicroelectronics.

Signed-off-by: Francois Ramu <francois.ramu@st.com>

FRASTM commented:
rebase

FRASTM commented:
rebase

galak commented on boards/arm/nucleo_g474re/arduino_r3_connector.dtsi:
```diff
@@ -0,0 +1,38 @@
+/ {
+	arduino_header: connector {
+		compatible = "arduino-header-r3";
+		#gpio-cells = <2>;
```
Need to add `gpio-map-mask` and `gpio-map-pass-thru` properties.

galak replied:
@erwango and/or @pabigot can help with how these should be set.

pabigot replied:
```
            gpio-map-mask = <0xffffffff 0xffffffc0>;
            gpio-map-pass-thru = <0 0x3f>;
```
for all cases.  See https://docs.zephyrproject.org/latest/guides/dts/index.html#gpio-nexus-nodes

FRASTM replied:
adding gpio-map-mask and gpio-map-pass-thru properties to boards/arm/nucleo_g474re/arduino_r3_connector.dtsi

galak reviewed (changes requested):
looks good other than the one dts update.

dbkinder reviewed (approved):
LGTM, thanks!

FRASTM committed:
boards: stm32: add nucleo board stm32g474re

This patch adds the nucleo_g474re board
         based on the stm32g474 from STMicroelectronics.

Signed-off-by: Francois Ramu <francois.ramu@st.com>

FRASTM commented:
rebase

FRASTM commented:
rebase
